### PR TITLE
cp: Use PathBuf instead of String for paths

### PR DESCRIPTION
**Pull request by tmccombs**
I think this should avoid unnecessarily validating utf-8, and avoids a few allocations.

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
Good idea! Could you add a small test for this with some invalid utf-8, so we don't regress?

**Comment by tmccombs** [comment]
Sure.

**Review by tertsdiepraam — APPROVED** [review]
I like it! There are just some clippy lints to fix.

**Comment by jfinkels** [comment]
Looks like some test failures on macos and windows

**Comment by tmccombs** [comment]
hmm, it appears that mac won't actually let you create a file with an invalid utf-8 file name, and from some quick research windows might be similar?  For now I've just restricted the tests to linux.

File: tests/by-util/test_cp.rs [hunk 1]
```diff
@@ -15,11 +15,14 @@ use std::os::unix::fs::PermissionsExt;
...
 #[cfg(not(windows))]
 use std::path::Path;
+#[cfg(target_os = "linux")]
+use std::path::PathBuf;
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use filetime::FileTime;
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use rlimit::Resource;
+use std::ffi::OsString;
```
Comment (sylvestre) [inline tests/by-util/test_cp.rs:25]: clippy says "unused import:"

File: tests/by-util/test_ls.rs [hunk 2]
```diff
@@ -814,8 +814,8 @@ fn test_ls_zero() {
     let scene = TestScenario::new(util_name!());
     let at = &scene.fixtures;
     at.mkdir("0-test-zero");
-    at.touch(&at.plus_as_string("2-test-zero"));
-    at.touch(&at.plus_as_string("3-test-zero"));
+    at.touch(at.plus_as_string("2-test-zero"));
```
Comment (sylvestre) [inline tests/by-util/test_ls.rs:817]: These changes seems to be unrelated. Can you move them into a different commit? or PR

Comment (tmccombs) [reply tests/by-util/test_ls.rs:817]: They are fixing clippy warnings because I changed the signature of `touch` to take `AsRef<Path>` instead of `&str`.

Comment (sylvestre) [reply tests/by-util/test_ls.rs:817]: oh, right, thanks

Comment (sylvestre) [reply tests/by-util/test_ls.rs:817]: could you please fix the other below ?
tests/by-util/test_chown.rs
test_install etc
